File: rectangulo.cpp
#include <iostream>
using namespace std;

class rectangulo {
private:
    int ancho;
    int longitud;

public:
    rectangulo(int _ancho, int _longitud);
    void mostrar_perimetro();
    void mostrar_area();
    void cambiar_ancho(int nuevo_ancho);
    void cambiar_longitud(int nueva_longitud);
};

rectangulo::rectangulo(int _ancho, int _longitud) {
    ancho = _ancho;
    longitud = _longitud;
}

void rectangulo::mostrar_perimetro() {
    int perimetro = (ancho * 2) + (longitud * 2);
    cout << "perimetro: " << perimetro << endl;
}

void rectangulo::mostrar_area() {
    int area = longitud * ancho;
    cout << "area: " << area << endl;
}

void rectangulo::cambiar_ancho(int nuevo_ancho) {
    ancho = nuevo_ancho;
}

void rectangulo::cambiar_longitud(int nueva_longitud) {
    longitud = nueva_longitud;
}

int main() {
    int ancho, longitud;
    cout << "ingrese ancho inicial: ";
    cin >> ancho;
    cout << "ingrese longitud inicial: ";
    cin >> longitud;

    rectangulo mi_rectangulo(ancho, longitud);

    char opcion;

    mi_rectangulo.mostrar_perimetro();
    mi_rectangulo.mostrar_area();

    do {
        cout << "que desea hacer? (a para cambiar ancho, b para cambiar longitud, c para salir): ";
        cin >> opcion;

        switch (opcion) {
            case 'a':
                cout << "ingrese nuevo ancho: ";
                cin >> ancho;
                mi_rectangulo.cambiar_ancho(ancho);
                mi_rectangulo.mostrar_perimetro();
                mi_rectangulo.mostrar_area();
                break;
            case 'b':
                cout << "ingrese nueva longitud: ";
                cin >> longitud;
                mi_rectangulo.cambiar_longitud(longitud);
                mi_rectangulo.mostrar_perimetro();
                mi_rectangulo.mostrar_area();
                break;
            case 'c':
                cout << "saliendo del programa." << endl;
                break;
            default:
                cout << "opcion no valida." << endl;
        }
    } while (opcion != 'c');

    return 0;
}

File: rectangulo.py
class Rectangulo:
    def __init__(self, ancho, longitud):
        self.ancho = ancho
        self.longitud = longitud

    def mostrar_perimetro(self):
        perimetro = (self.ancho * 2) + (self.longitud * 2)
        print("perimetro:", perimetro)

    def mostrar_area(self):
        area = self.longitud * self.ancho
        print("area:", area)

    def cambiar_ancho(self, nuevo_ancho):
        self.ancho = nuevo_ancho

    def cambiar_longitud(self, nueva_longitud):
        self.longitud = nueva_longitud

if __name__ == "__main__":
    ancho = int(input("ingrese el ancho inicial: "))
    longitud = int(input("ingrese la longitud inicial: "))

    mi_rectangulo = Rectangulo(ancho, longitud)

    opcion = ''

    mi_rectangulo.mostrar_perimetro()
    mi_rectangulo.mostrar_area()

    while opcion != 'c':
        print("¿que desea hacer? (a para cambiar ancho, b para cambiar longitud, c para salir): ")
        opcion = input()

        if opcion == 'a':
            nuevo_ancho = int(input("ingrese el nuevo ancho: "))
            mi_rectangulo.cambiar_ancho(nuevo_ancho)
            mi_rectangulo.mostrar_perimetro()
            mi_rectangulo.mostrar_area()
        elif opcion == 'b':
            nueva_longitud = int(input("ingrese la nueva longitud: "))
            mi_rectangulo.cambiar_longitud(nueva_longitud)
            mi_rectangulo.mostrar_perimetro()
            mi_rectangulo.mostrar_area()
        elif opcion == 'c':
            print("saliendo del programa.")
        else:
            print("opcion no valida.")
